modules/planning/tasks/deciders/open_space_decider/open_space_roi_decider.cc

```diff
@@ -449,17 +449,132 @@ void OpenSpaceRoiDecider::GetRoadBoundary(
+    double right_road_width = nearby_path.GetRoadRightWidth(check_point_s);
+
+    double current_road_width = std::max(left_road_width, right_road_width);
+
+    // get road boundaries at current location
+    common::PointENU check_point_xy;
+    std::vector<hdmap::RoadRoiPtr> road_boundaries;
+    std::vector<hdmap::JunctionInfoConstPtr> junctions;
+    check_point_xy.set_x(check_point.x());
+    check_point_xy.set_y(check_point.y());
+    hdmap_->GetRoadBoundaries(check_point_xy, current_road_width,
+                              &road_boundaries, &junctions);
+    hdmap::LineBoundary right_boundary;
+    hdmap::LineBoundary left_boundary;
+    if (check_point_s < center_line_s) {
+      right_boundary = (*road_boundaries.at(0)).left_boundary;
+      left_boundary = (*road_boundaries.at(0)).right_boundary;
+    } else {
+      right_boundary = (*road_boundaries.at(0)).right_boundary;
+      left_boundary = (*road_boundaries.at(0)).left_boundary;
+    }
+
+    // save road boundaries
+    for (size_t i = 0; i < left_boundary.line_points.size(); i++) {
+      left_lane_boundary->push_back(Vec2d(left_boundary.line_points[i].x(),
+                                          left_boundary.line_points[i].y()));
+    }
+    for (size_t i = 0; i < right_boundary.line_points.size(); i++) {
+      right_lane_boundary->push_back(Vec2d(right_boundary.line_points[i].x(),
+                                           right_boundary.line_points[i].y()));
+    }
+
+    center_lane_boundary_right->push_back(check_point);
+    center_lane_boundary_left->push_back(check_point);
+    center_lane_s_left->push_back(check_point_s);
+    center_lane_s_right->push_back(check_point_s);
+    left_lane_road_width->push_back(left_road_width);
+    right_lane_road_width->push_back(right_road_width);
+
+    if (check_point_s == end_s) {
+      break;
+    }
+    index += 10.0;
+    check_point_s = check_point_s + index;
+    check_point_s = check_point_s > end_s ? end_s : check_point_s;
+  }
+
+  size_t left_point_size = left_lane_boundary->size();
+  size_t right_point_size = right_lane_boundary->size();
+  ADEBUG << "right_road_boundary size: " << right_lane_boundary->size();
+  ADEBUG << "left_road_boundary size: " << left_lane_boundary->size();
+  for (size_t i = 0; i < left_point_size; i++) {
+    left_lane_boundary->at(i) -= origin_point;
+    left_lane_boundary->at(i).SelfRotate(-origin_heading);
+    ADEBUG << "left_road_boundary: [" << std::setprecision(9)
+           << left_lane_boundary->at(i).x() << ", "
+           << left_lane_boundary->at(i).y() << "]";
+  }
+  for (size_t i = 0; i < right_point_size; i++) {
+    right_lane_boundary->at(i) -= origin_point;
+    right_lane_boundary->at(i).SelfRotate(-origin_heading);
+    ADEBUG << "right_road_boundary: [" << std::setprecision(9)
+           << right_lane_boundary->at(i).x() << ", "
+           << right_lane_boundary->at(i).y() << "]";
+  }
+  if (!left_lane_boundary->empty()) {
```

Comment:
Double check if this is still needed when the search step is multiplier of point distance

Reply:
point distance is not a constant in hdmap, so it is difficult to set a search step base on point distance.